packager/media/file/udp_file.cc

```diff
@@ -6,25 +6,31 @@
 
 #include "packager/media/file/udp_file.h"
 
+#if defined(OS_WIN)
+#include <windows.h>
+#include <ws2tcpip.h>
+#include <memory>
+#define close closesocket
+#else
 #include <arpa/inet.h>
 #include <errno.h>
 #include <strings.h>
 #include <sys/socket.h>
 #include <unistd.h>
+#define INVALID_SOCKET -1
+#endif  // defined(OS_WIN)
 
 #include <limits>
 
 #include "packager/base/logging.h"
 #include "packager/media/file/udp_options.h"
 
-// TODO(tinskip): Adapt to work with winsock.
-
 namespace shaka {
 namespace media {
 
 namespace {
 
-const int kInvalidSocket(-1);
+const SOCKET kInvalidSocket(INVALID_SOCKET);
```

Comment:
You may use INVALID_SOCKET directly in the code below and remove the definition here.